dateformatterutil/src/test/java/com/example/dateformatter/DateFormatterUtilTest.kt

```diff
@@ -0,0 +1,114 @@
+package com.example.dateformatter
+
+import org.joda.time.DateTimeZone
+import org.joda.time.LocalDate
+import org.joda.time.LocalDateTime
+import org.joda.time.tz.UTCProvider
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+import java.util.*
+
+
+class DateFormatterUtilTest{
+
+    @Before
+    fun init(){
+        //this line solve the problem : ZoneInfoMap not found
+        DateTimeZone.setProvider(UTCProvider())
+    }
+    @Test
+    fun convertStringToDate_return_valid_day_month_year_when_passing_valid_param() {
+        //value returned is subtraction of 1900 according to documentation
+        assertEquals(
+            com.example.dateformatter.DateFormatterUtil.convertStringToDate(
+                "01.02.2022 01:32:27",
+                "dd.MM.yyyy HH:mm:ss"
+            )?.year!! + 1900, 2022
+        )
+        //value returned passed on enum in documentation
+        assertEquals(
+            com.example.dateformatter.DateFormatterUtil.convertStringToDate(
+                "12.01.2022 01:32:27",
+                "dd.MM.yyyy HH:mm:ss"
+            )?.day!!, 3
+        )
+        //value returned starts from 0 which represents january
+        assertEquals(
+            com.example.dateformatter.DateFormatterUtil.convertStringToDate(
+                "12.01.2022 01:32:27",
+                "dd.MM.yyyy HH:mm:ss"
+            )?.month!!, 0
+        )
+    }
+
+    @Test
+    fun convertDateToString_return_valid_String_when_passing_valid_param() {
+        assertEquals(
+            com.example.dateformatter.DateFormatterUtil.convertDateToString(
+                com.example.dateformatter.DateFormatterUtil.convertStringToDate(
+                    "01.02.2022 01:32:27",
+                    "dd.MM.yyyy HH:mm:ss"
+                )!!, "dd.MM.yyyy HH:mm:ss"
+            ), "01.02.2022 01:32:27"
+        )
+    }
+
+    @Test
+    fun convertToHijriDate() {
+        assertEquals(
+            com.example.dateformatter.DateFormatterUtil.convertToHijriDate(
+                com.example.dateformatter.DateFormatterUtil.convertStringToDate(
+                    "12.01.2022 01:32:27",
+                    "dd.MM.yyyy HH:mm:ss"
+                )!!
+            )?.year, LocalDate("1443-06-08").year
+        )
+    }
```

Comment:
The data of 12 / 01 /2021 should return 9 / 6 / 1443 in Higri not 8 / 6 / 1443 
Fix the function in your formatter class